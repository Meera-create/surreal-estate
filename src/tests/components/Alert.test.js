import React from 'react';
import { render } from '@testing-library/react';
import Alert from '../../components/alert';
import '@testing-library/jest-dom';

describe('tests alert component', () => {
    test('displays an error message', () => {
        const { getByText, asFragment } = render(<Alert message="Error!" />);
        expect(getByText(/Error/).textContent).toBe('Error!');
        expect(asFragment()).toMatchSnapshot();
    });
    test('displays success message', () => {
        const { getByText, asFragment } = render(<Alert message="Success!!" success={true} />);
        expect(getByText(/Success/).textContent).toBe('Success!!');
        expect(asFragment()).toMatchSnapshot();
    });
    test('doesnt render success message if message props is empty', () => {
        const { asFragment } = render(<Alert message="" />);
        const alert = asFragment();
        expect(alert).toMatchSnapshot();
    });
});


