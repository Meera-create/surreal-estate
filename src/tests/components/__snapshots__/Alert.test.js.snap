// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests alert component displays an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`tests alert component displays success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!
  </div>
</DocumentFragment>
`;

exports[`tests alert component doesnt render success message if message props is empty 1`] = `<DocumentFragment />`;
